Automattic/jetpack
# ci: Add project structure check

#### Changes proposed in this Pull Request:

Checks for the following things:
- projects/* and projects/*/* should be directories, no files other than
  maybe README.md, .gitignore, or .gitkeep.
- composer.json must exist.
- composer.json must include a monorepo `.repositories` entry.
- composer.json must require-dev (or just require) changelogger.
- Changelogger's changes-dir must have a .gitkeep.
- Changelogger's changes-dir must be production-excluded.
- .github/ must be export-ignored for packages.

#### Jetpack product discussion


None

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Is the new CI check happy?
* Create projects with `jetpack generate`, then run `.github/files/lint-project-structure.sh`. Does it report any errors?
